<launch>
  <!-- USB port name for dynamixel_port -->
  <arg name="port" default="/dev/ttyUSB0" />
  <arg name="load_robot_description" default="true" />

  <param if="$(arg load_robot_description)" name="robot_description"
    command="$(find xacro)/xacro --inorder '$(find crane_x7_description)/urdf/crane_x7.urdf.xacro'"
    />
  <group ns="crane_x7">
    <rosparam file="$(find crane_x7_control)/config/crane_x7_control.yaml" command="load"/>
    <param name="dynamixel_port/port_name" value="$(arg port)"/>

    <node name="controller_manager"
        pkg="controller_manager"
        type="spawner" respawn="false"
        output="screen"
        args="joint_state_controller
              arm_controller
              gripper_controller"/>

    <node name="crane_x7_control" pkg="crane_x7_control" type="crane_x7_control" output="screen" respawn="false">
      <!-- <param name="config_ns" value="/crane_x7"/> -->
    </node>

    <node name="robot_state_publisher" pkg="robot_state_publisher" 
        type="robot_state_publisher" respawn="false" output="screen">
      <remap from="/joint_states" to="/crane_x7/joint_states" />
    </node>
  </group>
</launch>
